src/sparseml/pytorch/utils/mfac_helpers.py

```diff
@@ -38,6 +50,75 @@
 ]
 
 
+class GradSampler:
+    """
+    Class for computing gradient samples for a Model given a sample data loader and
+    loss function.
+
+    :param dataloader: iterator of data samples to use as model inputs and their loss
+        targets. Samples can either be single tensors as model input or a list of
+        inputs and should be iterated in tuples with their targets
+    :param loss_fn: function to be called on model outputs to compute the loss at
+        each step
+    """
+
+    def __init__(
+        self,
+        dataloader: Iterator[Tuple[Union[Tensor, List[Tensor]], Any]],
+        loss_fn: Callable[[Tensor], Tensor],
+    ):
+        self._dataloader = dataloader
+        self._loss_fn = loss_fn
+
+    @staticmethod
+    def module_forward(module: Module, data: Union[Tensor, List[Tensor]]) -> Any:
+        """
+        :param module: module to perform forward pass with
+        :param data: single data sample to pass to module
+        :return: output(s) of the module forward pass
+        """
+        if isinstance(data, Tensor):
```

Comment:
would it make sense to use the tensors_module_forward function here ideally making it more robust and generic?